pkg/k8s/watchers/pod.go

```diff
@@ -99,6 +99,28 @@ func (k *K8sWatcher) createPodController(getter cache.Getter, fieldSelector fiel
 
 func (k *K8sWatcher) podsInit(k8sClient kubernetes.Interface, asyncControllers *sync.WaitGroup) {
 	var once sync.Once
+	watchNodePods := func() chan struct{} {
+		// Only watch for pod events for our node.
+		podStore, podController := k.createPodController(
+			k8sClient.CoreV1().RESTClient(),
+			fields.ParseSelectorOrDie("spec.nodeName="+nodeTypes.GetName()))
+		isConnected := make(chan struct{})
+		k.podStoreMU.Lock()
+		k.podStore = podStore
+		k.podStoreMU.Unlock()
+		k.podStoreOnce.Do(func() {
+			close(k.podStoreSet)
```

Comment:
Why was this not needed before?

Reply:
@tgraf this is required now because in CEP mode `watchNodePods` is executed a single time. When running in non-CEP mode this function can be executed multiple times.